tests/e2e/test_pgsql.py

```diff
@@ -0,0 +1,94 @@
+"""
+Module to perform PGSQL workload
+"""
+import logging
+import pytest
+import time
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating, utils
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, tier1
+from tests import helpers
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def ripsaw(request):
+    # Create Secret and Pool
+    secret = helpers.create_secret(constants.CEPHBLOCKPOOL)
+    pool = helpers.create_ceph_block_pool()
+
+    # Create storage class
+    log.info("Creating a Storage Class")
+    sc = helpers.create_storage_class(sc_name='pgsql-workload',
+                                      interface_type=constants.CEPHBLOCKPOOL,
+                                      secret_name=secret.name,
+                                      interface_name=pool.name)
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+
+    def teardown():
+        ripsaw.cleanup()
+        sc.delete()
+        secret.delete()
+        pool.delete()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@tier1
+class TestPgSQLWorkload(E2ETest):
+    """
+    Deploy an PGSQL workload using operator
+    """
+    def test_sql_workload_simple(self, ripsaw):
+        """
+        This is a basic pgsql workload
+        """
+        # Deployment postgres
+        log.info(
+            f"Deploying postgres database"
+        )
+        ripsaw.apply_crd('resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml')
+        ripsaw.setup_postgresql()
+        run_cmd(f'oc wait --for condition=ready pod '
+                f'-l app=postgres '
+                f'--timeout=120s')
+
+        # Create pgbench benchmark
+        log.info(
+            f"Create resource file for pgbench workload"
+        )
+        pg_data = templating.load_yaml_to_dict(constants.PGSQL_BENCHMARK_YAML)
+        pg_obj = OCS(**pg_data)
+        pg_obj.create()
+        # Wait for pgbench pod to be created
+        log.info(
+            f"waiting for pgbench benchmark to create, "
+            f"PGbench pod name: {pg_obj.name} "
+        )
+        wait_time = 120
+        log.info(f"Waiting {wait_time} seconds...")
+        time.sleep(wait_time)
```

Comment:
Instead, consider waiting for the pod to move to "completed" state.

Reply:
Fixed.  Added the check for pgbench_pod to move to Completed state.

Reply:
:+1:

Reply:
has this change been implemented yet?

Reply:
Yes, the 30 sec wait time for the pod to create.  Pod completion is checked as below:

run_cmd(
            f'bin/oc wait --for condition=Initialized '
            f'pods/{pgbench_pod} '
            f'--timeout=60s'
        )
        run_cmd(
            f'bin/oc wait --for condition=Complete jobs '
            f'-l app=pgbench-client '
            f'--timeout=300s'
        )